src/vs/workbench/contrib/terminalContrib/suggest/browser/terminalCompletionService.ts

```diff
@@ -559,6 +563,25 @@ function addPathRelativePrefix(text: string, resourceRequestConfig: Pick<Termina
 	return text;
 }
 
+/**
+ * Escapes special characters in a file/folder label for shell completion.
+ * This ensures that characters like [, ], etc. are properly escaped.
+ */
+export function escapeTerminalCompletionLabel(label: string, shellType: TerminalShellType, pathSeparator: string): string {
+	// Only escape for bash/zsh/fish; PowerShell and cmd have different rules
+	if (shellType === GeneralShellType.PowerShell || shellType === WindowsShellType.CommandPrompt) {
+		// TODO: Implement escaping for PowerShell/cmd if needed
+		return label;
+	}
+
+	const specialCharsSet = new Set(['[', ']', '(', ')', '\'', '"', '\\', '`', '*', '?', ';', '&', '|', '<', '>']);
+	const specialCharsRegex = /[\[\]\(\)'"\\\`\*\?;|&<>]/;
+	if (!specialCharsRegex.test(label)) {
+		return label;
+	}
+	return label.split('').map(c => specialCharsSet.has(c) ? '\\' + c : c).join('');
```

Comment:
Can simplify and use just regex:

```ts
input.replace(/[\[\]\(\)'"\\\`\*\?;|&<>]/g, '\\$&');
```

![image](https://github.com/user-attachments/assets/ee0385c3-2e46-472a-a109-8f59f283654c)